sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoPushService.kt

```diff
@@ -43,10 +44,33 @@ open class KlaviyoPushService : FirebaseMessagingService() {
     override fun onMessageReceived(message: RemoteMessage) {
         super.onMessageReceived(message)
 
-        if (message.isKlaviyoNotification) {
-            KlaviyoNotification(message).displayNotification(this)
+        if (message.isKlaviyoMessage) {
+            if (message.isKlaviyoNotification) {
+                KlaviyoNotification(message).displayNotification(this)
+            } else {
```

Comment:
Ooh wait, correct me if I'm wrong, but a notification could also be bearing key/value pairs right? So this logic can't be exclusively if/else? Rather it'd need to be (pseudocode)
```
if (message.isKlaviyoNotification){
  //display
}
if (message.hasKeyValuePairs) {
  //handle k/v pairs
}
``` 
?

Reply:
Also, if that isn't the case _now_ but it _could be_ the case in the future, we should still design the SDK for that eventuality i believe.